sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/SearchOptions.java

```diff
@@ -949,22 +949,34 @@ public SearchOptions setSemanticFields(List<String> semanticFields) {
     }
 
     /**
-     * Get the vector property: The query parameters for vector and hybrid search queries.
+     * Set the vectors property using a single vector: The query parameters for vector and hybrid search
+     * queries.
+     *
+     * @param vector the vector value to set.
+     * @return the SearchRequest object itself.
+     */
+    public SearchOptions setVector(SearchQueryVector vector) {
```

Comment:
Do we typically add the singular overload in Java when we're also doing var args?

Reply:
We have a few different patterns, but this is indeed not one of them. Depending on the use case, a few valid combinations would include:

```
setVectors(List<SearchQueryVector> vectors)
setVectors(SearchQueryVector... vectors)
```

and

```
setVectors(List<SearchQueryVector> vectors)
addVectors(SearchQueryVector... vectors)
```

More often than not the first pattern is used, with the set calls clearly indicating complete replacement of all SearchQueryVectors, and avoiding the less useful semantics of 'adding' more vectors to an existing set that the second pattern allows. However, as noted, it depends on the expected use case from customers.

I'll also add: I looked at the [published JavaDocs](https://learn.microsoft.com/en-us/java/api/com.azure.search.documents.models.searchoptions?view=azure-java-stable) and did not see any existing vector API. Therefore, I'll ignore the 'breaking' comments that I would raise otherwise, as I don't think they apply here, but please let me know if I'm missing something.

Reply:
`setVector` can be dropped in favor of the varargs overload we have for `setVectors`.

Reply:
@JonathanGiles I'll change it to follow the first pattern. As far as the JavaDocs go, everything in main for Search is part of the public preview and not GA. GA bug fixes and patch releases get done in branches off the release tags.